### DISCUSSION TITLE
4GB Bar on 4GB VRAM card breaks DirectX

### DISCUSSION THREAD

**Discussion by MrAnonymous7122** [xCuri0/ReBarUEFI#200] [General]
@xCuri0 
As seen in issue 11, I managed to get a full 4GB bar on a Sapphire Nitro+ RX 580 4GB.
However as the title states, DirectX and DirectML no longer work.
![5](https://github.com/xCuri0/ReBarUEFI/assets/161169713/3ff8c27a-ae5f-4fd8-8f52-f974e1dc9126)
I've tried reinstalling DirectX and it hasn't fixed it. 
A lot of applications don't work and give DirectX errors.
GPU-Z is also unable to give information on DirectX 9,10,11 saying "Could not initialise"
For DX12. it says device not found.
I will respond tomorrow.

**Comment by MrAnonymous7122** [comment]
Resume from sleep also doesn't work but that's not a major issue.

**Comment by xCuri0** [comment]
This is a known issue which is why I say for 4GB VRAM Polaris users to use 2GB BAR as a buggy workaround.

Only AMD can fix this it's an issue with their Windows drivers, @romulus2k4 asked AMD about this and they said they have no plans to fix this issue.

4GB BAR on 4GB Polaris works fine on Linux though so that's an option.


You could also try using DXVK on Windows if Vulkan is working but still it would be broken and unusable.

**Reply by MrAnonymous7122** [reply]
So it does technically work but breaks directx, ok.
I will try DXVK, in hope that it does work.
Thanks for your help.

**Reply by MrAnonymous7122** [reply]
I cant even compile DXVK, terminal needs directx to work.

**Reply by 2k1Sisi** [reply]
Is Resizable BAR enabled in your AMD Adrenaline? mine is enabled in GPU-Z but not available in Adrenaline
